projects/packages/sync/src/class-actions.php

```diff
@@ -507,6 +505,35 @@ public static function do_initial_sync() {
 		self::do_full_sync( $initial_sync_config );
 	}
 
+	/**
+	 * Whether the site should do an initial sync.
+	 *
+	 * An initial Sync should not be done under these circumstances:
+	 *  - A full sync is currently in progress.
+	 *  - The site is not registering and an initial sync has already been started.
+	 *
+	 * @return bool Whether to run an initial sync.
+	 */
+	public static function should_do_initial_sync() {
+		$full_sync_module = Modules::get_module( 'full-sync' );
+
+		if ( ! $full_sync_module ) {
+			return false;
+		}
+
+		// Don't start a new sync if a full sync is in progress.
+		if ( $full_sync_module->is_started() && ! $full_sync_module->is_finished() ) {
+			return false;
+		}
+
+		// Don't start a new sync if the site isn't registering and an initial sync was already started.
+		if ( ! doing_action( 'jetpack_site_registered' ) && $full_sync_module->is_started() ) {
```

Comment:
@kbrown9 can you outline why this check on 'jetpack_site_registered' is added? Logically it does make sense but it is a departure from existing functionality where the action 'jetpack_user_authorized' can also trigger do_initial_sync.

Also won't this check on is_started cause issues with sites that delete Jetpack than re-install?  

Would it make sense to add flag of $strict or similar that would preserve existing flow and then add this if statement only if strict is true? This way WC Pay can call `do_initial_sync( true )` and get the flow they want but not change existing flows?

Reply:
I like your proposal to add a flag. I think an even better approach would be to introduce a new method, `Actions::do_only_first_initial_sync`, which consumer plugins could call when they want an initial sync run only when one has never been run before.

The `Actions::do_intial_sync` method is hooked to a few actions, and the `jetpack_site_registered` action passes a [few arguments](https://github.com/Automattic/jetpack/blob/1d03317f5a562093fe0c4214fdf27092b3d092f9/projects/packages/connection/src/class-manager.php#L1180) to the callback functions. We specify that `Actions::do_initial_sync` accepts 0 arguments from the action hook [here](https://github.com/Automattic/jetpack/blob/532ea71e3da11560fe4c4874457ea9201bab30de/projects/packages/sync/src/class-main.php#L35), but if that ever needed to change, I think we would run into problems.

I think introducing the new `Actions::do_only_first_initial_sync` method will be both clearer and safer than adding a flag to the existing `Actions::do_intial_sync` method. What do you think of this approach?

Reply:
I agree a new method is cleaner and safer for usage. 👍 from me on this approach